shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt

```diff
@@ -255,6 +257,27 @@ class StepQuizReducer(
                 } else {
                     null
                 }
+            is Message.TheoryToolbarClicked -> {
+                if (state.stepQuizState is StepQuizState.AttemptLoaded && state.stepQuizState.isTheoryAvailable) {
+                    val topicTheoryId = state.stepQuizState.step.topicTheory
+
+                    if (topicTheoryId != null) {
+                        state to setOf(
+                            Action.LogAnalyticEvent(
+                                StepQuizClickedTheoryToolbarHyperskillAnalyticEvent(
+                                    stepRoute.analyticRoute,
+                                    topicTheoryId
+                                )
+                            ),
+                            Action.ViewAction.NavigateTo.StepScreen(StepRoute.Learn(topicTheoryId))
```

Comment:
Asked Egor about do we need to persist current step route (create a copy of the current one but with a different id) or we can always use learn route

Reply:
We need to create a copy of the step route with a new id
https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-758/Shared-study-plan-opens-current-practice-step-and-doesnt-show-theory#focus=Comments-74-233083.0-0